core/plugins/org.fusesource.ide.foundation.core/src/org/fusesource/ide/foundation/core/util/JobWaiterUtil.java

```diff
@@ -77,4 +84,9 @@ public void setEndless(boolean isEndless) {
 		this.isEndless = isEndless;
 	}
 	
+	public static void updateUI() {
+		while (Display.getDefault().readAndDispatch()) {
```

Comment:
this is really an ugly and dangerous hack.
Please provide recording of the added improvement.
Even if you don't find the root cause, please don't loop indefinitely and loop for a limited amount time

Reply:
Without call to updateUI
https://drive.google.com/open?id=14qikA94YaDQhLjyNp7mS4EzjpJccOLUB

With call to updateUI
https://drive.google.com/open?id=1wGWWKq3G8kGNvq8wEgrFM3lNgqOk0D1a